sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -157,6 +165,27 @@ _rom_ext_start_boot:
   la   t0, (_rom_ext_interrupt_vector + 1)
   csrw mtvec, t0
 
+  /**
+   * Initialize the Ibex simple address translation to run position dependent
+   * code on the virtual flash space.
+   */
+  .extern _flash_start
+  .extern _vflash_start
+  .extern _vflash_size
+  la a0, _vflash_start
+  la a1, _vflash_size
+  la a2, _flash_start
```

Comment:
I can see how this might work currently but I think it is going to be problematic to be using the absolute start address of flash for this. I think we probably need to avoid using symbols and instead just check which flash slot the current code is in by loading the current address and comparing it to the known slot addresses. Essentially doing something like this pseudocode:

```
if instruction_addr >= slot_b_addr
  then
    source_addr=slot_b_addr
  else
    source_addr=slot_a_addr
```

Reply:
I choose a branchless approach by computing the current slot using the `pc`, entry_point address and vflash address:
```
addr = pc - _vflash_address - _rom_ext_start_boot
```